### DISCUSSION TITLE
Colouring Fibre with a bumpKernel

### DISCUSSION THREAD

**Discussion by ctriant1** [mtex-toolbox/mtex#1059] [Ask Anything] [answered]
Hi all,

Hope everyone is keeping safe and well. As per the title, I often need to colour certain fibres from EBSD data (usually certain planes say within 15 degrees of the loading axis) and in previous versions I was able to use a bumpKernel to define the range. While S2DeLaValleePoussin has its merits, I sometimes need to set that hard angle limit as I'm not concerned about which grains are included within that range as opposed to how close to the target vector they are.

I can currently run this with no issues and produces the expected result

`% define a fibre - Basal`
`f = fibre(Miller(0,0,1,cs),zvector);`
`% set up coloring`
`colorKey = ipfSpotKey(cs);`
`colorKey.inversePoleFigureDirection = f.r;`
`colorKey.center = f.h;`
`colorKey.color = [0 0 1];`
`colorKey.psi = S2DeLaValleePoussin('halfwidth',HW*degree);`
`figure`
`plot(ebsdPX('Titanium (Alpha)'),colorKey.orientation2color(ebsdPX('Titanium (Alpha)').orientations))`

If I replace colorKey.psi with

`colorKey.psi = bumpKernel(HW*degree);`

the colorKey is constructed but when trying to plot with the same syntax as before, I encounter the following issue

`No appropriate method, property, or field 'eval' for class 'bumpKernel'.`

Running MATLAB 2018a with MTEX 5.6.0. Also to note that I had definitely used bumpKernel to colour fibres in an older MTEX version, so I'm assuming something needs to be changed in the plot settings unless bumpKernel has been dropped.

**Comment by kilir** [comment, marked as answer]
Hi,
there's the `'S2BumpKernel'` which should do does what you are looking for.
Cheers, Rüdiger

**Reply by ctriant1** [reply]
Hi Rüdiger,
thank you for the quick suggestion! Any idea why I'm getting `Undefined function or variable 'S2BumpKernel'.` ?

**Reply by kilir** [reply]
Your mtex version is too old, it's quite recent. See here: https://github.com/mtex-toolbox/mtex/commit/d15351bf65a059c1fb3dddc32a908a80d5a18684

**Reply by ctriant1** [reply]
That is right, I've updated to 5.7.0 and I can confirm that it works! Many thanks, Chris
